sdk/ai/azure-ai-inference/azure/ai/inference/_client.py

```diff
@@ -0,0 +1,90 @@
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# Code generated by Microsoft (R) Python Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is regenerated.
+# --------------------------------------------------------------------------
+
+from copy import deepcopy
+from typing import Any
+
+from azure.core import PipelineClient
+from azure.core.credentials import AzureKeyCredential
+from azure.core.pipeline import policies
+from azure.core.rest import HttpRequest, HttpResponse
+
+from ._configuration import ModelClientConfiguration
+from ._operations import ModelClientOperationsMixin
+from ._serialization import Deserializer, Serializer
+
+
+class ModelClient(ModelClientOperationsMixin):  # pylint: disable=client-accepts-api-version-keyword
+    """ModelClient.
+
+    :param credential: Credential needed for the client to connect to Azure. Required.
+    :type credential: ~azure.core.credentials.AzureKeyCredential
+    :keyword endpoint: Service host. Required.
+    :paramtype endpoint: str
+    :keyword api_version: The API version to use for this operation. Default value is
+     "2024-04-01-preview". Note that overriding this default value may result in unsupported
+     behavior.
+    :paramtype api_version: str
+    """
+
+    def __init__(self, credential: AzureKeyCredential, *, endpoint: str, **kwargs: Any) -> None:
```

Comment:
We should support more than key credentials, no?

Reply:
Yes, eventually, but at the moment only key credentials is supported. Entra ID will be supported later, not sure when. I will try to get clarifications on the timeline.